bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -0,0 +1,258 @@
+            String requestContent = String.format(LOGIN_BODY_BUILDER,
+                    Base64.getEncoder().encodeToString("realm=particuliers".getBytes(StandardCharsets.UTF_8)),
+                    config.username, config.password);
+            InputStream stream = new ByteArrayInputStream(requestContent.getBytes(StandardCharsets.UTF_8));
+            logger.debug("executeUrl POST {} requestContent {}", LOGIN_BASE_URI, requestContent);
+            HttpUtil.executeUrl("POST", LOGIN_BASE_URI, stream, "application/x-www-form-urlencoded",
+                    HTTP_DEFAULT_TIMEOUT_MS);
+            stream.close();
+
+            // first call to test connection and to get needed cookies
+            // getEnedisInfo(DAILY, LocalDate.now(), LocalDate.now());
+
+            updateStatus(ThingStatus.ONLINE);
+            refreshJob = scheduler.scheduleWithFixedDelay(() -> {
+                updateLinkyData();
+            }, 0, config.refreshInterval, TimeUnit.MINUTES);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+        }
+    }
+
+    /**
+     * Request new data and updates channels
+     */
+    private void updateLinkyData() {
+        final LocalDate today = LocalDate.now();
+
+        LocalDate rangeStart = today.minusDays(9);
+        EnedisInfo result = getEnedisInfo(DAILY, rangeStart, today);
+        if (result != null && result.success()) {
+            int jump = result.getDecalage();
+            while (rangeStart.getDayOfWeek() != weekFields.getFirstDayOfWeek()) {
+                rangeStart = rangeStart.plusDays(1);
+                jump++;
+            }
+
+            int lastWeekNumber = rangeStart.get(weekFields.weekOfWeekBasedYear());
+
+            double lastWeek = 0;
+            double thisWeek = 0;
+            double yesterday = -1;
+            while (jump < result.getData().size()) {
+                double consumption = result.getData().get(jump).valeur;
+                if (consumption > 0) {
+                    if (rangeStart.get(weekFields.weekOfWeekBasedYear()) == lastWeekNumber) {
+                        lastWeek += consumption;
+                    } else {
+                        thisWeek += consumption;
+                    }
+                    yesterday = consumption;
+                }
+                jump++;
+                rangeStart = rangeStart.plusDays(1);
+            }
+
+            updateKwhChannel(YESTERDAY, yesterday);
+            updateKwhChannel(THIS_WEEK, thisWeek);
+            updateKwhChannel(LAST_WEEK, lastWeek);
+        }
+
+        double lastMonth = -1;
+        double thisMonth = -1;
+        result = getEnedisInfo(MONTHLY, today.withDayOfMonth(1).minusMonths(1), today);
+        if (result != null && result.success()) {
+            lastMonth = result.getData().stream().filter(EnedisInfo.Data::isPositive).findFirst().get().valeur;
```

Comment:
Why not doing it like you do for the yearly values ?

Shouldn't you rather use the `order` information included in data ?